### ISSUE TITLE
Assertion error in ExpandedMapSeqOrOption.scala:138

### ISSUE THREAD

**Issue by Sciss** [Sciss/Lucre#34]
In GitLab by @Sciss on Jun 12, 2020, 20:43

Seeing this in Thru Segm:
```
---> mezzo-spth
mezzo-spth: db? false, pool? true
---> mezzo-rsmp
mezzo-rsmp: begin (391s)
---> proc-lk
proc-plk: pool-in exhausted
---> proc-paul
proc-paul: begin (75s)
---> proc-rsmp-c1
proc-rsmp-c1: pool-in exhausted
---> rec
rec-fill: run rec (120s)
late 0.148306499
rec-mock: 0.00500143
---> mezzo-spth
mezzo-spth: db? false, pool? true
---> proc-lk
proc-plk: pool-in exhausted
---> proc-rsmp-c1
proc-rsmp-c1: pool-in exhausted
---> mezzo-spth
mezzo-spth: db? false, pool? true
---> proc-lk
proc-plk: pool-in exhausted
---> proc-rsmp-c1
proc-rsmp-c1: pool-in exhausted
loudM: 18
java.lang.AssertionError: assertion failed
	at scala.Predef$.assert(Predef.scala:267)
	at de.sciss.lucre.expr.graph.impl.ExpandedMapSeqLike.buildResult(ExpandedMapSeqOrOption.scala:138)
	at de.sciss.lucre.expr.graph.impl.ExpandedMapSeqLike.buildResult(ExpandedMapSeqOrOption.scala:123)
	at de.sciss.lucre.expr.graph.impl.ExpandedMapSeqOrOption.pullChange(ExpandedMapSeqOrOption.scala:74)
	at de.sciss.lucre.expr.graph.impl.ExpandedMapSeqOrOption.pullChange(ExpandedMapSeqOrOption.scala:26)
	at de.sciss.lucre.event.IChangeEvent.pullUpdate(IChangeEvent.scala:22)
	at de.sciss.lucre.event.IChangeEvent.pullUpdate$(IChangeEvent.scala:21)
	at de.sciss.lucre.expr.graph.impl.ExpandedMapSeqOrOption.pullUpdate(ExpandedMapSeqOrOption.scala:26)
	at de.sciss.lucre.event.IPush$Impl.apply(IPush.scala:197)
	at de.sciss.lucre.event.IPush$Impl.$anonfun$pull$1(IPush.scala:151)
	at scala.collection.Iterator$$anon$10.nextCur(Iterator.scala:583)
	at scala.collection.Iterator$$anon$10.hasNext(Iterator.scala:597)
	at scala.collection.mutable.ListBuffer.addAll(ListBuffer.scala:118)
	at scala.collection.mutable.ListBuffer$.from(ListBuffer.scala:38)
	at scala.collection.immutable.List$.from(List.scala:617)
	at scala.collection.IterableOnceOps.toList(IterableOnce.scala:1234)
	at scala.collection.IterableOnceOps.toList$(IterableOnce.scala:1234)
	at scala.collection.AbstractIterator.toList(Iterator.scala:1274)
	at de.sciss.lucre.event.IPush$Impl.pull(IPush.scala:147)
	at de.sciss.lucre.event.IPush$.apply(IPush.scala:33)
	at de.sciss.lucre.event.impl.IGenerator.fire(IGenerator.scala:22)
	at de.sciss.lucre.event.impl.IGenerator.fire$(IGenerator.scala:20)
	at de.sciss.lucre.expr.graph.Attr$WithDefault$Expanded.fire(Attr.scala:236)
	at de.sciss.lucre.expr.graph.Attr$WithDefault$Expanded.$anonfun$obsAttr$2(Attr.scala:251)
	at de.sciss.lucre.expr.graph.Attr$WithDefault$Expanded.$anonfun$obsAttr$2$adapted(Attr.scala:245)
	at de.sciss.lucre.expr.impl.CellViewImpl$OptionOrElseImpl.$anonfun$react$9(CellViewImpl.scala:236)
	at de.sciss.lucre.expr.impl.CellViewImpl$OptionOrElseImpl.$anonfun$react$9$adapted(CellViewImpl.scala:236)
	at de.sciss.lucre.expr.graph.impl.ObjCellViewVarImpl$Observation.$anonfun$mkDurObs$2(ObjCellViewVarImpl.scala:77)
	at de.sciss.lucre.expr.graph.impl.ObjCellViewVarImpl$Observation.$anonfun$mkDurObs$2$adapted(ObjCellViewVarImpl.scala:75)
	at de.sciss.lucre.event.Observer$Impl.apply(Observer.scala:54)
	at de.sciss.lucre.event.Observer$Impl.apply(Observer.scala:43)
	at de.sciss.lucre.event.Push$Reaction.$anonfun$apply$1(Push.scala:40)
	at de.sciss.lucre.event.Push$Reaction.$anonfun$apply$1$adapted(Push.scala:40)
	at scala.collection.immutable.List.foreach(List.scala:305)
	at de.sciss.lucre.event.Push$Reaction.apply(Push.scala:40)
	at de.sciss.lucre.event.Push$Impl.$anonfun$pull$3(Push.scala:96)
	at de.sciss.lucre.event.Push$Impl.$anonfun$pull$3$adapted(Push.scala:96)
	at scala.collection.immutable.List.foreach(List.scala:305)
	at de.sciss.lucre.event.Push$Impl.pull(Push.scala:96)
	at de.sciss.lucre.event.Push$.apply(Push.scala:29)
	at de.sciss.lucre.event.impl.Generator.fire(Generator.scala:22)
	at de.sciss.lucre.event.impl.Generator.fire$(Generator.scala:20)
	at de.sciss.lucre.stm.impl.ListImpl$Impl$changed$.fire(ListImpl.scala:133)
	at de.sciss.lucre.stm.impl.ListImpl$Impl.fireAdded(ListImpl.scala:213)
	at de.sciss.lucre.stm.impl.ListImpl$Impl.insert(ListImpl.scala:199)
	at de.sciss.lucre.stm.impl.ListImpl$Impl.addLast(ListImpl.scala:165)
	at de.sciss.lucre.stm.impl.ListImpl$Impl.addLast(ListImpl.scala:92)
	at de.sciss.lucre.edit.EditFolder$.de$sciss$lucre$edit$EditFolder$$appendDo(EditFolder.scala:112)
	at de.sciss.lucre.edit.EditFolder$.$anonfun$append$1(EditFolder.scala:25)
	at scala.runtime.java8.JFunction0$mcV$sp.apply(JFunction0$mcV$sp.scala:18)
	at scala.Option.fold(Option.scala:263)
	at de.sciss.lucre.edit.EditFolder$.append(EditFolder.scala:26)
	at de.sciss.lucre.expr.graph.Folder$AppendExpanded.$anonfun$executeAction$1(Folder.scala:273)
	at de.sciss.lucre.expr.graph.Folder$AppendExpanded.$anonfun$executeAction$1$adapted(Folder.scala:270)
	at scala.Option.foreach(Option.scala:437)
	at de.sciss.lucre.expr.graph.Folder$AppendExpanded.executeAction(Folder.scala:270)
	at de.sciss.lucre.expr.graph.Folder$AppendExpanded.executeAction(Folder.scala:265)
	at de.sciss.lucre.expr.graph.Act$ExpandedSeq.$anonfun$executeAction$1(Act.scala:25)
	at de.sciss.lucre.expr.graph.Act$ExpandedSeq.$anonfun$executeAction$1$adapted(Act.scala:25)
	at scala.collection.IterableOnceOps.foreach(IterableOnce.scala:553)
	at scala.collection.IterableOnceOps.foreach$(IterableOnce.scala:551)
	at scala.collection.AbstractIterable.foreach(Iterable.scala:921)
	at de.sciss.lucre.expr.graph.Act$ExpandedSeq.executeAction(Act.scala:25)
	at de.sciss.lucre.expr.graph.Act$ExpandedSeq.executeAction(Act.scala:23)
	at de.sciss.lucre.expr.graph.ThenAct$Expanded.loop$4(IfElse.scala:251)
	at de.sciss.lucre.expr.graph.ThenAct$Expanded.executeIfDefined(IfElse.scala:260)
	at de.sciss.lucre.expr.graph.ThenAct$Expanded.executeIfDefined(IfElse.scala:239)
	at de.sciss.lucre.expr.graph.ElseAct$Expanded.executeAction(IfElse.scala:288)
	at de.sciss.lucre.expr.graph.ElseAct$Expanded.executeAction(IfElse.scala:284)
	at de.sciss.lucre.expr.graph.ThenAct$Expanded.loop$4(IfElse.scala:251)
	at de.sciss.lucre.expr.graph.ThenAct$Expanded.executeIfDefined(IfElse.scala:260)
	at de.sciss.lucre.expr.graph.ThenAct$Expanded.executeIfDefined(IfElse.scala:239)
	at de.sciss.lucre.expr.graph.ElseAct$Expanded.executeAction(IfElse.scala:288)
	at de.sciss.lucre.expr.graph.ElseAct$Expanded.executeAction(IfElse.scala:284)
	at de.sciss.lucre.expr.impl.IActionImpl.$anonfun$addSource$2(IActionImpl.scala:32)
	at de.sciss.lucre.expr.impl.IActionImpl.$anonfun$addSource$2$adapted(IActionImpl.scala:32)
	at de.sciss.lucre.event.Observer$IImpl.apply(Observer.scala:37)
	at de.sciss.lucre.event.IPush$Reaction.$anonfun$apply$1(IPush.scala:44)
	at de.sciss.lucre.event.IPush$Reaction.$anonfun$apply$1$adapted(IPush.scala:44)
	at scala.collection.immutable.List.foreach(List.scala:305)
	at de.sciss.lucre.event.IPush$Reaction.apply(IPush.scala:44)
	at de.sciss.lucre.event.IPush$Impl.$anonfun$pull$3(IPush.scala:156)
	at de.sciss.lucre.event.IPush$Impl.$anonfun$pull$3$adapted(IPush.scala:156)
	at scala.collection.immutable.List.foreach(List.scala:305)
	at de.sciss.lucre.event.IPush$Impl.pull(IPush.scala:156)
	at de.sciss.lucre.event.IPush$.apply(IPush.scala:33)
	at de.sciss.lucre.event.impl.IGenerator.fire(IGenerator.scala:22)
	at de.sciss.lucre.event.impl.IGenerator.fire$(IGenerator.scala:20)
	at de.sciss.lucre.expr.graph.Runner$ExpandedState.fire(Runner.scala:84)
	at de.sciss.lucre.expr.graph.Runner$ExpandedState.$anonfun$obs$2(Runner.scala:94)
	at de.sciss.lucre.expr.graph.Runner$ExpandedState.$anonfun$obs$2$adapted(Runner.scala:90)
	at de.sciss.lucre.event.impl.ObservableImpl.$anonfun$fire$1(ObservableImpl.scala:31)
	at de.sciss.lucre.event.impl.ObservableImpl.$anonfun$fire$1$adapted(ObservableImpl.scala:31)
	at scala.collection.IterableOnceOps.foreach(IterableOnce.scala:553)
	at scala.collection.IterableOnceOps.foreach$(IterableOnce.scala:551)
	at scala.collection.AbstractIterable.foreach(Iterable.scala:921)
	at de.sciss.lucre.event.impl.ObservableImpl.fire(ObservableImpl.scala:31)
	at de.sciss.lucre.event.impl.ObservableImpl.fire$(ObservableImpl.scala:29)
	at de.sciss.fscape.lucre.impl.FScapeRunnerImpl$Impl.fire(FScapeRunnerImpl.scala:51)
	at de.sciss.synth.proc.impl.BasicViewBaseImpl.state_$eq(BasicViewBaseImpl.scala:37)
	at de.sciss.synth.proc.impl.BasicViewBaseImpl.state_$eq$(BasicViewBaseImpl.scala:33)
	at de.sciss.fscape.lucre.impl.FScapeRunnerImpl$Impl.state_$eq(FScapeRunnerImpl.scala:51)
	at de.sciss.fscape.lucre.impl.FScapeRunnerImpl$Impl.$anonfun$run$4(FScapeRunnerImpl.scala:135)
	at de.sciss.fscape.lucre.impl.FScapeRunnerImpl$Impl.$anonfun$run$4$adapted(FScapeRunnerImpl.scala:131)
	at de.sciss.lucre.event.impl.ObservableImpl.$anonfun$fire$1(ObservableImpl.scala:31)
	at de.sciss.lucre.event.impl.ObservableImpl.$anonfun$fire$1$adapted(ObservableImpl.scala:31)
	at scala.collection.IterableOnceOps.foreach(IterableOnce.scala:553)
	at scala.collection.IterableOnceOps.foreach$(IterableOnce.scala:551)
	at scala.collection.AbstractIterable.foreach(Iterable.scala:921)
	at de.sciss.lucre.event.impl.ObservableImpl.fire(ObservableImpl.scala:31)
	at de.sciss.lucre.event.impl.ObservableImpl.fire$(ObservableImpl.scala:29)
	at de.sciss.fscape.lucre.impl.RenderingImpl$Impl.fire(RenderingImpl.scala:279)
	at de.sciss.fscape.lucre.impl.RenderingImpl$Impl.$anonfun$completeWith$1(RenderingImpl.scala:333)
	at de.sciss.fscape.lucre.impl.RenderingImpl$Impl.$anonfun$completeWith$1$adapted(RenderingImpl.scala:320)
	at de.sciss.lucre.confluent.impl.CursorImpl$Impl.performStep(CursorImpl.scala:137)
	at de.sciss.lucre.confluent.impl.CursorImpl$Impl.$anonfun$stepTag$1(CursorImpl.scala:121)
	at de.sciss.lucre.confluent.impl.CursorImpl$Impl.$anonfun$topLevelAtomic$1(CursorImpl.scala:113)
	at scala.concurrent.stm.ccstm.InTxnImpl.runBlock(InTxnImpl.scala:570)
	at scala.concurrent.stm.ccstm.InTxnImpl.topLevelAttempt(InTxnImpl.scala:526)
	at scala.concurrent.stm.ccstm.InTxnImpl.topLevelAtomicImpl(InTxnImpl.scala:398)
	at scala.concurrent.stm.ccstm.InTxnImpl.atomic(InTxnImpl.scala:261)
	at scala.concurrent.stm.ccstm.CCSTMExecutor.apply(CCSTMExecutor.scala:24)
	at de.sciss.lucre.stm.Txn$.atomic(Txn.scala:108)
	at de.sciss.lucre.confluent.impl.CursorImpl$Impl.topLevelAtomic(CursorImpl.scala:111)
	at de.sciss.lucre.confluent.impl.CursorImpl$Impl.stepTag(CursorImpl.scala:119)
	at de.sciss.lucre.confluent.impl.CursorImpl$Impl.step(CursorImpl.scala:116)
	at de.sciss.synth.proc.SoundProcesses$.$anonfun$step$1(SoundProcesses.scala:91)
	at scala.runtime.java8.JFunction0$mcV$sp.apply(JFunction0$mcV$sp.scala:18)
	at scala.concurrent.Future$.$anonfun$apply$1(Future.scala:661)
	at scala.concurrent.impl.Promise$Transformation.run(Promise.scala:430)
	at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
	at java.base/java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:304)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
	at java.base/java.lang.Thread.run(Thread.java:834)
From 'FScape completeWith'
java.lang.AssertionError: assertion failed
	at scala.Predef$.assert(Predef.scala:267)
	at de.sciss.lucre.expr.graph.impl.ExpandedMapSeqLike.buildResult(ExpandedMapSeqOrOption.scala:138)
	at de.sciss.lucre.expr.graph.impl.ExpandedMapSeqLike.buildResult(ExpandedMapSeqOrOption.scala:123)
	at de.sciss.lucre.expr.graph.impl.ExpandedMapSeqOrOption.pullChange(ExpandedMapSeqOrOption.scala:74)
	at de.sciss.lucre.expr.graph.impl.ExpandedMapSeqOrOption.pullChange(ExpandedMapSeqOrOption.scala:26)
	at de.sciss.lucre.event.IChangeEvent.pullUpdate(IChangeEvent.scala:22)
	at de.sciss.lucre.event.IChangeEvent.pullUpdate$(IChangeEvent.scala:21)
	at de.sciss.lucre.expr.graph.impl.ExpandedMapSeqOrOption.pullUpdate(ExpandedMapSeqOrOption.scala:26)
	at de.sciss.lucre.event.IPush$Impl.apply(IPush.scala:197)
	at de.sciss.lucre.event.IPush$Impl.$anonfun$pull$1(IPush.scala:151)
	at scala.collection.Iterator$$anon$10.nextCur(Iterator.scala:583)
	at scala.collection.Iterator$$anon$10.hasNext(Iterator.scala:597)
	at scala.collection.mutable.ListBuffer.addAll(ListBuffer.scala:118)
	at scala.collection.mutable.ListBuffer$.from(ListBuffer.scala:38)
	at scala.collection.immutable.List$.from(List.scala:617)
	at scala.collection.IterableOnceOps.toList(IterableOnce.scala:1234)
	at scala.collection.IterableOnceOps.toList$(IterableOnce.scala:1234)
	at scala.collection.AbstractIterator.toList(Iterator.scala:1274)
	at de.sciss.lucre.event.IPush$Impl.pull(IPush.scala:147)
	at de.sciss.lucre.event.IPush$.apply(IPush.scala:33)
	at de.sciss.lucre.event.impl.IGenerator.fire(IGenerator.scala:22)
	at de.sciss.lucre.event.impl.IGenerator.fire$(IGenerator.scala:20)
	at de.sciss.lucre.expr.graph.Attr$WithDefault$Expanded.fire(Attr.scala:236)
	at de.sciss.lucre.expr.graph.Attr$WithDefault$Expanded.$anonfun$obsAttr$2(Attr.scala:251)
	at de.sciss.lucre.expr.graph.Attr$WithDefault$Expanded.$anonfun$obsAttr$2$adapted(Attr.scala:245)
	at de.sciss.lucre.expr.impl.CellViewImpl$OptionOrElseImpl.$anonfun$react$9(CellViewImpl.scala:236)
	at de.sciss.lucre.expr.impl.CellViewImpl$OptionOrElseImpl.$anonfun$react$9$adapted(CellViewImpl.scala:236)
	at de.sciss.lucre.expr.graph.impl.ObjCellViewVarImpl$Observation.$anonfun$mkDurObs$2(ObjCellViewVarImpl.scala:77)
	at de.sciss.lucre.expr.graph.impl.ObjCellViewVarImpl$Observation.$anonfun$mkDurObs$2$adapted(ObjCellViewVarImpl.scala:75)
	at de.sciss.lucre.event.Observer$Impl.apply(Observer.scala:54)
	at de.sciss.lucre.event.Observer$Impl.apply(Observer.scala:43)
	at de.sciss.lucre.event.Push$Reaction.$anonfun$apply$1(Push.scala:40)
	at de.sciss.lucre.event.Push$Reaction.$anonfun$apply$1$adapted(Push.scala:40)
	at scala.collection.immutable.List.foreach(List.scala:305)
	at de.sciss.lucre.event.Push$Reaction.apply(Push.scala:40)
	at de.sciss.lucre.event.Push$Impl.$anonfun$pull$3(Push.scala:96)
	at de.sciss.lucre.event.Push$Impl.$anonfun$pull$3$adapted(Push.scala:96)
	at scala.collection.immutable.List.foreach(List.scala:305)
	at de.sciss.lucre.event.Push$Impl.pull(Push.scala:96)
	at de.sciss.lucre.event.Push$.apply(Push.scala:29)
	at de.sciss.lucre.event.impl.Generator.fire(Generator.scala:22)
	at de.sciss.lucre.event.impl.Generator.fire$(Generator.scala:20)
	at de.sciss.lucre.stm.impl.ListImpl$Impl$changed$.fire(ListImpl.scala:133)
	at de.sciss.lucre.stm.impl.ListImpl$Impl.fireAdded(ListImpl.scala:213)
	at de.sciss.lucre.stm.impl.ListImpl$Impl.insert(ListImpl.scala:199)
	at de.sciss.lucre.stm.impl.ListImpl$Impl.addLast(ListImpl.scala:165)
	at de.sciss.lucre.stm.impl.ListImpl$Impl.addLast(ListImpl.scala:92)
	at de.sciss.lucre.edit.EditFolder$.de$sciss$lucre$edit$EditFolder$$appendDo(EditFolder.scala:112)
	at de.sciss.lucre.edit.EditFolder$.$anonfun$append$1(EditFolder.scala:25)
	at scala.runtime.java8.JFunction0$mcV$sp.apply(JFunction0$mcV$sp.scala:18)
	at scala.Option.fold(Option.scala:263)
	at de.sciss.lucre.edit.EditFolder$.append(EditFolder.scala:26)
	at de.sciss.lucre.expr.graph.Folder$AppendExpanded.$anonfun$executeAction$1(Folder.scala:273)
	at de.sciss.lucre.expr.graph.Folder$AppendExpanded.$anonfun$executeAction$1$adapted(Folder.scala:270)
	at scala.Option.foreach(Option.scala:437)
	at de.sciss.lucre.expr.graph.Folder$AppendExpanded.executeAction(Folder.scala:270)
	at de.sciss.lucre.expr.graph.Folder$AppendExpanded.executeAction(Folder.scala:265)
	at de.sciss.lucre.expr.graph.Act$ExpandedSeq.$anonfun$executeAction$1(Act.scala:25)
	at de.sciss.lucre.expr.graph.Act$ExpandedSeq.$anonfun$executeAction$1$adapted(Act.scala:25)
	at scala.collection.IterableOnceOps.foreach(IterableOnce.scala:553)
	at scala.collection.IterableOnceOps.foreach$(IterableOnce.scala:551)
	at scala.collection.AbstractIterable.foreach(Iterable.scala:921)
	at de.sciss.lucre.expr.graph.Act$ExpandedSeq.executeAction(Act.scala:25)
	at de.sciss.lucre.expr.graph.Act$ExpandedSeq.executeAction(Act.scala:23)
	at de.sciss.lucre.expr.graph.ThenAct$Expanded.loop$4(IfElse.scala:251)
	at de.sciss.lucre.expr.graph.ThenAct$Expanded.executeIfDefined(IfElse.scala:260)
	at de.sciss.lucre.expr.graph.ThenAct$Expanded.executeIfDefined(IfElse.scala:239)
	at de.sciss.lucre.expr.graph.ElseAct$Expanded.executeAction(IfElse.scala:288)
	at de.sciss.lucre.expr.graph.ElseAct$Expanded.executeAction(IfElse.scala:284)
	at de.sciss.lucre.expr.graph.ThenAct$Expanded.loop$4(IfElse.scala:251)
	at de.sciss.lucre.expr.graph.ThenAct$Expanded.executeIfDefined(IfElse.scala:260)
	at de.sciss.lucre.expr.graph.ThenAct$Expanded.executeIfDefined(IfElse.scala:239)
	at de.sciss.lucre.expr.graph.ElseAct$Expanded.executeAction(IfElse.scala:288)
	at de.sciss.lucre.expr.graph.ElseAct$Expanded.executeAction(IfElse.scala:284)
	at de.sciss.lucre.expr.impl.IActionImpl.$anonfun$addSource$2(IActionImpl.scala:32)
	at de.sciss.lucre.expr.impl.IActionImpl.$anonfun$addSource$2$adapted(IActionImpl.scala:32)
	at de.sciss.lucre.event.Observer$IImpl.apply(Observer.scala:37)
	at de.sciss.lucre.event.IPush$Reaction.$anonfun$apply$1(IPush.scala:44)
	at de.sciss.lucre.event.IPush$Reaction.$anonfun$apply$1$adapted(IPush.scala:44)
	at scala.collection.immutable.List.foreach(List.scala:305)
	at de.sciss.lucre.event.IPush$Reaction.apply(IPush.scala:44)
	at de.sciss.lucre.event.IPush$Impl.$anonfun$pull$3(IPush.scala:156)
	at de.sciss.lucre.event.IPush$Impl.$anonfun$pull$3$adapted(IPush.scala:156)
	at scala.collection.immutable.List.foreach(List.scala:305)
	at de.sciss.lucre.event.IPush$Impl.pull(IPush.scala:156)
	at de.sciss.lucre.event.IPush$.apply(IPush.scala:33)
	at de.sciss.lucre.event.impl.IGenerator.fire(IGenerator.scala:22)
	at de.sciss.lucre.event.impl.IGenerator.fire$(IGenerator.scala:20)
	at de.sciss.lucre.expr.graph.Runner$ExpandedState.fire(Runner.scala:84)
	at de.sciss.lucre.expr.graph.Runner$ExpandedState.$anonfun$obs$2(Runner.scala:94)
	at de.sciss.lucre.expr.graph.Runner$ExpandedState.$anonfun$obs$2$adapted(Runner.scala:90)
	at de.sciss.lucre.event.impl.ObservableImpl.$anonfun$fire$1(ObservableImpl.scala:31)
	at de.sciss.lucre.event.impl.ObservableImpl.$anonfun$fire$1$adapted(ObservableImpl.scala:31)
	at scala.collection.IterableOnceOps.foreach(IterableOnce.scala:553)
	at scala.collection.IterableOnceOps.foreach$(IterableOnce.scala:551)
	at scala.collection.AbstractIterable.foreach(Iterable.scala:921)
	at de.sciss.lucre.event.impl.ObservableImpl.fire(ObservableImpl.scala:31)
	at de.sciss.lucre.event.impl.ObservableImpl.fire$(ObservableImpl.scala:29)
	at de.sciss.fscape.lucre.impl.FScapeRunnerImpl$Impl.fire(FScapeRunnerImpl.scala:51)
	at de.sciss.synth.proc.impl.BasicViewBaseImpl.state_$eq(BasicViewBaseImpl.scala:37)
	at de.sciss.synth.proc.impl.BasicViewBaseImpl.state_$eq$(BasicViewBaseImpl.scala:33)
	at de.sciss.fscape.lucre.impl.FScapeRunnerImpl$Impl.state_$eq(FScapeRunnerImpl.scala:51)
	at de.sciss.fscape.lucre.impl.FScapeRunnerImpl$Impl.$anonfun$run$4(FScapeRunnerImpl.scala:135)
	at de.sciss.fscape.lucre.impl.FScapeRunnerImpl$Impl.$anonfun$run$4$adapted(FScapeRunnerImpl.scala:131)
	at de.sciss.lucre.event.impl.ObservableImpl.$anonfun$fire$1(ObservableImpl.scala:31)
	at de.sciss.lucre.event.impl.ObservableImpl.$anonfun$fire$1$adapted(ObservableImpl.scala:31)
	at scala.collection.IterableOnceOps.foreach(IterableOnce.scala:553)
	at scala.collection.IterableOnceOps.foreach$(IterableOnce.scala:551)
	at scala.collection.AbstractIterable.foreach(Iterable.scala:921)
	at de.sciss.lucre.event.impl.ObservableImpl.fire(ObservableImpl.scala:31)
	at de.sciss.lucre.event.impl.ObservableImpl.fire$(ObservableImpl.scala:29)
	at de.sciss.fscape.lucre.impl.RenderingImpl$Impl.fire(RenderingImpl.scala:279)
	at de.sciss.fscape.lucre.impl.RenderingImpl$Impl.$anonfun$completeWith$1(RenderingImpl.scala:333)
	at de.sciss.fscape.lucre.impl.RenderingImpl$Impl.$anonfun$completeWith$1$adapted(RenderingImpl.scala:320)
	at de.sciss.lucre.confluent.impl.CursorImpl$Impl.performStep(CursorImpl.scala:137)
	at de.sciss.lucre.confluent.impl.CursorImpl$Impl.$anonfun$stepTag$1(CursorImpl.scala:121)
	at de.sciss.lucre.confluent.impl.CursorImpl$Impl.$anonfun$topLevelAtomic$1(CursorImpl.scala:113)
	at scala.concurrent.stm.ccstm.InTxnImpl.runBlock(InTxnImpl.scala:570)
	at scala.concurrent.stm.ccstm.InTxnImpl.topLevelAttempt(InTxnImpl.scala:526)
	at scala.concurrent.stm.ccstm.InTxnImpl.topLevelAtomicImpl(InTxnImpl.scala:398)
	at scala.concurrent.stm.ccstm.InTxnImpl.atomic(InTxnImpl.scala:261)
	at scala.concurrent.stm.ccstm.CCSTMExecutor.apply(CCSTMExecutor.scala:24)
	at de.sciss.lucre.stm.Txn$.atomic(Txn.scala:108)
	at de.sciss.lucre.confluent.impl.CursorImpl$Impl.topLevelAtomic(CursorImpl.scala:111)
	at de.sciss.lucre.confluent.impl.CursorImpl$Impl.stepTag(CursorImpl.scala:119)
	at de.sciss.lucre.confluent.impl.CursorImpl$Impl.step(CursorImpl.scala:116)
	at de.sciss.synth.proc.SoundProcesses$.$anonfun$step$1(SoundProcesses.scala:91)
	at scala.runtime.java8.JFunction0$mcV$sp.apply(JFunction0$mcV$sp.scala:18)
	at scala.concurrent.Future$.$anonfun$apply$1(Future.scala:661)
	at scala.concurrent.impl.Promise$Transformation.run(Promise.scala:430)
	at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
	at java.base/java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:304)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
	at java.base/java.lang.Thread.run(Thread.java:834)
```

Looking into the code, it seems this happens after an observed folder (Attr) was previously emptied. It looks like the `ref` still has the old content (e.g. three children). My guess from initial look is that the shortcut in line 68 `if (isEmpty(inV)) emptyOut` is causing the bug. Because that would indeed prevent the update of `ref` in the `now` phase.

**Comment by Sciss** [comment]
In GitLab by @Sciss on Jun 12, 2020, 21:51

Still persists after removing that conditional. Next hunch is we do not call into `IPush.tryPull` in `value`, which AFAIR is needed when using `Caching`?

**Comment by Sciss** [comment]
In GitLab by @Sciss on Jun 13, 2020, 24:20

fixing `value` doesn't solve it (yet -- although it was probably also wrong). Next idea is that we are still dispatching to a disposed object, because in `dispose` we set the reference to `empty`.

**Comment by Sciss** [comment]
In GitLab by @Sciss on Jun 13, 2020, 04:44

mentioned in commit SoundProcesses@37e9683ecc930d62c1158af6023d421a5c66f52c

**Comment by Sciss** [comment]
In GitLab by @Sciss on Jun 13, 2020, 16:55

mentioned in commit 5907206271f7f41e99254f4f8e8401150d9dd3b8

**Comment by Sciss** [comment]
In GitLab by @Sciss on Jun 13, 2020, 16:55

mentioned in commit 16c76cff5a51c1306096dd63710c4dff691d2ecb

**Comment by Sciss** [comment]
In GitLab by @Sciss on Jun 13, 2020, 16:55

closed via commit 16c76cff5a51c1306096dd63710c4dff691d2ecb
